src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -21,47 +71,598 @@ const ActorInit En_Box_InitVars = {
+        }
+        func_8002AA44(globalCtx, &pos, &vel, &accel, 150);
+    }
+}
+
+void EnBox_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnBox* this = THIS;
+
+    if (this->movementFlags & ENBOX_MOVE_STICK_TO_GROUND) {
+        this->movementFlags &= ~ENBOX_MOVE_STICK_TO_GROUND;
+        EnBox_ClipToGround(this, globalCtx);
+    }
+
+    (this->actionFunc)(this, globalCtx);
+
+    if (!(this->movementFlags & ENBOX_MOVE_IMMOBILE)) {
+        Actor_MoveForward(&this->dyna.actor);
+        func_8002E4B4(globalCtx, &this->dyna.actor, 0.0f, 0.0f, 0.0f, 0x1C);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/EnBox_Init.s")
+    switch (this->type) {
+        case ENBOX_TYPE_SMALL:
+        case ENBOX_TYPE_6:
+        case ENBOX_TYPE_ROOM_CLEAR_SMALL:
+        case ENBOX_TYPE_SWITCH_FLAG_FALL_SMALL:
+            Actor_SetHeight(&this->dyna.actor, 20.0f);
+            break;
+        default:
+            Actor_SetHeight(&this->dyna.actor, 40.0f);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/EnBox_Destroy.s")
+    if ((this->dyna.actor.params >> 5 & 0x7F) == 0x7C && this->actionFunc == EnBox_Open &&
+        this->skelanime.animCurrentFrame > 45 && this->iceSmokeTimer < 100) {
+        func_809C9EF8(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/func_809C92F4.s")
+void EnBox_PostLimbDraw2(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
+    EnBox* this = THIS;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/func_809C93F8.s")
+    if (limbIndex == 1) {
+        gSPMatrix((*gfx)++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_box.c", 1492),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        if (this->type != ENBOX_TYPE_DECORATED_BIG) {
+            gSPDisplayList((*gfx)++, D_060006F0);
+        } else {
+            gSPDisplayList((*gfx)++, D_06000AE8);
+        }
+    } else if (limbIndex == 3) {
+        gSPMatrix((*gfx)++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_box.c", 1502),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        if (this->type != ENBOX_TYPE_DECORATED_BIG) {
+            gSPDisplayList((*gfx)++, D_060010C0);
+        } else {
+            gSPDisplayList((*gfx)++, D_06001678);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/func_809C94AC.s")
+Gfx* EnBox_EmptyDList(GraphicsContext* gfxCtx) {
+    Gfx* dList;
+    Gfx* dListHead;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/func_809C9630.s")
+    dListHead = Graph_Alloc(gfxCtx, sizeof(Gfx));
+    if (dListHead == NULL) {
+        __assert("gfxp != NULL", "../z_en_box.c", 1528);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Box/func_809C9700.s")
+    dList = dListHead;
+    gSPEndDisplayList(dList++);
```

Comment:
It's the other way around, we call `dList` the pointer to the generated display list, while `dListHead` is a pointer to the head of the list we're currently appending to.
```suggestion
    dListHead = dList;
    gSPEndDisplayList(dListHead++);
```